src/main/java/me/ryanhamshire/GriefPrevention/DataStore.java

```diff
@@ -1264,7 +1264,12 @@ synchronized void tryExtendSiege(Player player, Claim claim)
         if (claim.isAdminClaim()) return;
 
         //player must have some level of permission to be sieged in a claim
-        if (claim.allowAccess(player) != null) return;
```

Comment:
DataStore#tryExtendSiege is called during siege handling of container access. While using the new method technically wouldn't create an infinite loop, it would fire a second event, which seems like a bad idea overall. This does, however, change the behavior of siege so that the player being sieged does have to actually have permission in the claim instead of access via public trust.

Reply:
In v17 we're removing siege, but as long as siege works as intended for now it'll be fine.

Reply:
Siege is almost identical, the only change here is that siege no longer rolls in claims that the sieged user has trust in via `/trust public` - my reasoning was that it would be bad to re-call the same event funnel method, and since both users have the same level of access neither has an unfair advantage.